editors/code/src/inlay_hints.ts

```diff
@@ -39,11 +45,7 @@ export function activateInlayHints(ctx: Ctx) {
     maybeUpdater.onConfigChange().catch(console.error);
 }
 
-const typeHints = createHintStyle("type");
-const paramHints = createHintStyle("parameter");
-const chainingHints = createHintStyle("chaining");
-
-function createHintStyle(hintKind: "type" | "parameter" | "chaining") {
+function createHintStyle(ctx: Ctx, hintKind: "type" | "parameter" | "chaining"): InlayHintStyle {
```

Comment:
> ctx: Ctx

Feels a bit odd to drag the whole `Ctx` into the function to use a single param out of it.
I propose to pass a single, properly named boolean there instead or a hints' config at most.